pennylane/ops/op_math/pow_class.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that indicates the power of an operator.
+"""
+from copy import copy
+from scipy.linalg import fractional_matrix_power
+
+from pennylane.operation import (
+    DecompositionUndefinedError,
+    SparseMatrixUndefinedError,
+    PowUndefinedError,
+    Operator,
+    Operation,
+    Observable,
+)
+from pennylane.queuing import QueuingContext, apply
+
+from pennylane import math as qmlmath
+
+_superscript = str.maketrans("0123456789.+-", "⁰¹²³⁴⁵⁶⁷⁸⁹⋅⁺⁻")
+
+
+# pylint: disable=no-member
+class PowOperation(Operation):
+    """Operation-specific methods and properties for the ``Pow`` class.
+
+    Dynamically mixed in based on the provided base matrix.
+
+    """
+
+    # until we add gradient support
+    grad_method = None
+
+    def inv(self):
+        self.hyperparameters["z"] *= -1
+        self._name = f"{self.base.name}**{self.z}"
+        return self
+
+    @property
+    def inverse(self):
+        return False
+
+    @inverse.setter
+    def inverse(self, boolean):
+        if boolean is True:
+            raise NotImplementedError("The inverse can not be set for a power operator")
```

Comment:
The `inv` method changes the sign of the exponent rather than updating the `_inverse` property.  This way we can keep `op.inverse=False`.

Being able to set `inverse` is inconsistent with this. For example, if we tried:
```
op = Pow(qml.S(0), -1)
op.inverse=True
op.inverse=True
```

Would we flip the sign of the exponents both times?

I'm trying to set up a system that is still mostly compatible with our in-place inversion ecosystem, but at the same time is setting everything up for its deprecation. So all this inversion stuff is a temporary patch of functionality.